src/pages/pool/_id.vue

```diff
@@ -130,6 +131,7 @@ function addIntersectionObserver(): void {
 }
 onMounted(() => {
   addIntersectionObserver();
+  metaService.setOgImage(poolId);
```

Comment:
I don't see how this would work? The HTML structure is already rendered at this point? Would you not need to do this in the router?

Reply:
hm, seems problem is deeper, og scraper doesn't run js at all. putting this in router doesn't help

Reply:
yeah. seems we need to implement middleware in cloudflare to handle this